src/models/step_arguments/data_table_spec.js

```diff
@@ -44,6 +44,141 @@ describe('DataTable', function () {
     })
   })
 
+  describe('typed hashes', function() {
+    it('rejects tables where not all rows have 3 columns', function() {
+      const dataTable = new DataTable({
+        rows: [
+          {
+            cells: [
+              {value: 'key1'},
+              {value: 'string'},
+              {value:  'value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key2'},
+              {value: 'string'},
+              {value: 'value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key3'},
+              {value: 'value'}
+            ]
+          }
+        ]
+      })
+      expect(() => {
+        dataTable.typedRowsHash()
+      }).to.throw(Error)
+    })
+
+    it('rejects tables where not all types are recognized', function() {
+      const dataTable = new DataTable({
+        rows: [
+          {
+            cells: [
+              {value: 'key1'},
+              {value: 'string'},
+              {value:  'value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key2'},
+              {value: 'OUPS'},
+              {value: 'value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key3'},
+              {value: 'string'},
+              {value: 'value'}
+            ]
+          }
+        ]
+      })
+      expect(() => {
+        dataTable.typedRowsHash()
+      }).to.throw(Error)
+    })
+
+    it('accepts all supported types, case insensitive, no type given - is string', function() {
```

Comment:
Can we please break this test up into each type similar to what you did with date.